catalog/spatial/registry/registry-schema-bindings/src/main/java/org/codice/ddf/registry/schemabindings/RegistryPackageUtils.java

```diff
@@ -0,0 +1,279 @@
+    }
+
+    public static List<AssociationType1> getAssociations(
+            RegistryObjectListType registryObjectList) {
+        return getObjectsFromRegistryObjectList(registryObjectList, AssociationType1.class);
+    }
+
+    public static <T extends RegistryObjectType> List<T> getObjectsFromRegistryObjectList(
+            RegistryPackageType registryPackage, Class<T> type) {
+        List<T> registryObjects = new ArrayList<>();
+
+        if (registryPackage == null) {
+            return registryObjects;
+        }
+
+        if (registryPackage.isSetRegistryObjectList()) {
+            registryObjects =
+                    getObjectsFromRegistryObjectList(registryPackage.getRegistryObjectList(), type);
+        }
+
+        return registryObjects;
+    }
+
+    public static <T extends RegistryObjectType> List<T> getObjectsFromRegistryObjectList(
+            RegistryObjectListType registryObjectList, Class<T> type) {
+        List<T> registryObjects = new ArrayList<>();
+
+        if (registryObjectList == null) {
+            return registryObjects;
+        }
+
+        registryObjects.addAll(registryObjectList.getIdentifiable()
+                .stream()
+                .filter(identifiable -> type.isInstance(identifiable.getValue()))
+                .map(identifiable -> (T) identifiable.getValue())
+                .collect(Collectors.toList()));
+
+        return registryObjects;
+    }
+
+    public static SlotType1 getSlotByName(String name, List<SlotType1> slots) {
+        for (SlotType1 slot : slots) {
+            if (slot.getName()
+                    .equals(name)) {
+                return slot;
+            }
+        }
+
+        return null;
+    }
+
+    public static Map<String, SlotType1> getNameSlotMap(List<SlotType1> slots) {
+        Map<String, SlotType1> slotMap = new HashMap<>();
+
+        for (SlotType1 slot : slots) {
+            slotMap.put(slot.getName(), slot);
+        }
+        return slotMap;
+    }
+
+    public static Map<String, List<SlotType1>> getNameSlotMapDuplicateSlotNamesAllowed(
+            List<SlotType1> slots) {
+        Map<String, List<SlotType1>> slotMap = new HashMap<>();
+
+        for (SlotType1 slot : slots) {
+            slotMap.putIfAbsent(slot.getName(), new ArrayList<>());
+            slotMap.get(slot.getName())
+                    .add(slot);
+        }
+
+        return slotMap;
+    }
+
+    public static List<String> getSlotStringValues(SlotType1 slot) {
+        List<String> slotAttributes = new ArrayList<>();
+
+        if (slot.isSetValueList()) {
+            ValueListType valueList = slot.getValueList()
+                    .getValue();
+            if (valueList.isSetValue()) {
+                slotAttributes = valueList.getValue();
+            }
+        }
+
+        return slotAttributes;
+    }
+
+    public static List<Date> getSlotDateValues(SlotType1 slot) {
+        List<Date> dates = new ArrayList<>();
+
+        if (slot.isSetValueList()) {
```

Comment:
Can't we use `getSlotStringValues()` here and use `stream()` to replace the `for` loop? Also, I don't think that `Date.from()` can ever return `null`.